h2o-algos/src/main/java/hex/util/LinearAlgebraUtils.java

```diff
@@ -123,6 +220,50 @@ else if (!dinfo._catMissing[col])
     }
     return eigenvectors;
   }
+  
+  public static class FindMaxIndex extends MRTask<FindMaxIndex> {
+    public long _maxIndex = -1;
+    int _colIndex;
+    double _maxValue;
+    
+    public FindMaxIndex(int colOfInterest, double maxValue) {
+      _colIndex = colOfInterest;
+      _maxValue = maxValue;
+    }
+    
+    @Override
+    public void map(Chunk[] cs) {
+      int rowLen = cs[0].len();
+      long startRowIndex = cs[0].start();
+      for (int rowIndex=0; rowIndex < rowLen; rowIndex++) {
+        double rowVal = cs[_colIndex].atd(rowIndex);
+        if (rowVal == _maxValue) {
+          _maxIndex = startRowIndex+rowIndex;
+        }
+      }
+    }
+    
+    @Override public void reduce(FindMaxIndex other) {
+      if (this._maxIndex < 0)
+        this._maxIndex = other._maxIndex;
+      else if (this._maxIndex > other._maxIndex)
+        this._maxIndex = other._maxIndex; 
+    }
+  }
+  
+  public static class CopyQtoQMatrix extends MRTask<CopyQtoQMatrix> {
+    @Override public void map(Chunk[] cs) {
+      int totColumn = cs.length;  // all columns in cs.
+      int halfColumn = totColumn/2; // start of Q matrix
+      int totRows = cs[0].len();
```

Comment:
Are we 100% sure all chunks have the same length ? As this is not guaranteed for some Frames. I'm only asking, don't have deep insight ATM.

Reply:
MRTasks can only run on frames that have the same layout => all chunks do have the same length.